core/src/main/java/smile/neighbor/KDTree.java

```diff
@@ -373,7 +373,7 @@ private void search(double[] q, Node node, double radius, List<Neighbor<double[]
             search(q, nearer, radius, neighbors);
 
             // now look in further half
-            if (radius >= diff * diff) {
+            if (radius >= Math.abs(diff)) {
```

Comment:
but when the `abs(diff) > 1`, the original check might be wrong:
![image](https://cloud.githubusercontent.com/assets/4106365/25064747/754d6f5a-2209-11e7-8476-a01183bc7bbb.png)